### PR TITLE
DB integration

### PR DESCRIPTION
**Pull request by AxelMatstoms**
**Changes:**
* Add database models for user, dashboard and diagram.
* User model supports login of users

**I have:**

- [x] Set target branch to the correct branch, usually `dev`.
- [x] Ensured code is formatted, linted.
- [x] Cleaned up git history.
- [x] Ensured commit messages describe *what* was changed, and *why*.
- [x] Ran tests locally, checked output.

### CODE REVIEW

File: src/dashboard/models/user.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    ObjectIdField,
+    StringField,
+    ReferenceField
+)
+
+
+class Diagram(EmbeddedDocument):
+    """Diagram database mode.
+
+    A diagram stores all the information needed to reconstruct a Dast
```
Comment (fallensach) [inline src/dashboard/models/user.py:17]: Typo?

**Review by YouKnowBlom — APPROVED** [review]

**Review by haholm — APPROVED** [review]
Looks great! Good job.

**Review by TobiasElf — CHANGES_REQUESTED** [review]
Mostly looks good. Some comments about comments mostly

File: src/dashboard/models/data.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+class DataType(Enum):
+    """Data type enum."""
+
+    NUMERIC = "numeric"
+    XY_PLOT = "xy_plot"
+    ARRAY = "array"
+
+
+# TODO: to polars df
```
[the remark this answers is not in the document]
Comment (AxelMatstoms) [reply src/dashboard/models/data.py:75]: #139

File: src/dashboard/models/user.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    user = User(username=username)
+    user.save()
+
+    return user
+
+
+def login_user(username: str) -> User:
+    """Login as a user.
+
+    Logins as a user, creating the user if it does not exists.
```
[the remark this answers is not in the document]
Comment (AxelMatstoms) [reply src/dashboard/models/user.py:90]: My understanding is that this is expected behavior. A login system is not a requirement, so we don't want to spend time writing code for storing passwords etc.

File: src/dashboard/models/data.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    x_unit: str | None = None
+    y_unit: str | None = None
+    markers: list[Marker] = field(default_factory=list)
+
+    def __post_init__(self) -> None:
+        """Post initialize xy data.
+
+        Convert marker dicts to Marker objects.
+        """
+        self.markers = [Marker(**marker) for marker in self.markers]  # type: ignore
```
[the remark this answers is not in the document]
Comment (AxelMatstoms) [reply src/dashboard/models/data.py:217]: Possible, but requires another field to store the list of dicts. Would be kind of ugly imo.

Resolved by adding an inline comment explaining this.

File: src/dashboard/models/db.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+DB_URL_ENV_NAME = "DB_URL"
+USER_DB_NAME = "dashboard"
+
+
+def _get_db_url() -> str:
+    """Find the db url from the environment."""
+    db_url = os.getenv(DB_URL_ENV_NAME)
+    if not db_url:
+        raise EnvironmentError(f"Can't find environment variable {DB_URL_ENV_NAME}")
```
[the remark this answers is not in the document]
Comment (AxelMatstoms) [reply src/dashboard/models/db.py:27]: :+1:

File: src/dashboard/models/user.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+def register_user(username: str) -> User:
+    """Register a new user.
+
+    Creates a new user with a given username.
+
+    Args:
+        username (str): the username of the new user.
+
+    Returns:
+        A User object returning the newly created user.
```
[the remark this answers is not in the document]
Comment (AxelMatstoms) [reply src/dashboard/models/user.py]: Yes, this was a typo.

**Review by TobiasElf — APPROVED** [review]
LGTM!

**Pull request merged**